bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java

```diff
@@ -0,0 +1,796 @@
+                service.closeApp(launchSession, listener);
+                break;
+            /*
+             * case Media:
+             * if (service instanceof MediaPlayer) {
+             * ((MediaPlayer) service).closeMedia(launchSession, listener);
+             * }
+             * break;
+             *
+             *
+             * case WebApp:
+             * if (service instanceof WebAppLauncher) {
+             * ((WebAppLauncher) service).closeWebApp(launchSession, listener);
+             * }
+             * break;
+             * case Unknown:
+             */
+            default:
+                listener.onError("This DeviceService does not know ho to close this LaunchSession");
+                break;
+        }
+    }
+
+    public void closeApp(LaunchSession launchSession, ResponseListener<CommandConfirmation> listener) {
+        String uri = "ssap://system.launcher/close";
+        String appId = launchSession.getAppId();
+        String sessionId = launchSession.getSessionId();
+
+        JsonObject payload = new JsonObject();
+        payload.addProperty("id", appId);
+        payload.addProperty("sessionId", sessionId);
+
+        ServiceCommand<CommandConfirmation> request = new ServiceCommand<>(uri, payload,
+                x -> GSON.fromJson(x, CommandConfirmation.class), listener);
+        launchSession.getService().sendCommand(request);
+    }
+
+    public ServiceSubscription<AppInfo> subscribeRunningApp(ResponseListener<AppInfo> listener) {
+        ServiceSubscription<AppInfo> request = new ServiceSubscription<>(FOREGROUND_APP, null,
+                jsonObj -> GSON.fromJson(jsonObj, AppInfo.class), listener);
+        sendCommand(request);
+        return request;
+
+    }
+
+    // KEYBOARD
+
+    public ServiceSubscription<TextInputStatusInfo> subscribeTextInputStatus(
+            ResponseListener<TextInputStatusInfo> listener) {
+        return keyboardInput.connect(listener);
+    }
+
+    public void sendText(String input) {
+        keyboardInput.sendText(input);
+    }
+
+    public void sendEnter() {
+        keyboardInput.sendEnter();
+    }
+
+    public void sendDelete() {
+        keyboardInput.sendDel();
+    }
+
+    // MOUSE
+
+    public void executeMouse(Consumer<LGWebOSTVMouseSocket> onConnected) {
+        LGWebOSTVMouseSocket mouseSocket = new LGWebOSTVMouseSocket(this.client);
+        mouseSocket.setListener(new WebOSTVMouseSocketListener() {
+
+            @Override
+            public void onStateChanged(LGWebOSTVMouseSocket.State oldState, LGWebOSTVMouseSocket.State newState) {
+                switch (newState) {
+                    case CONNECTED:
+                        onConnected.accept(mouseSocket);
+                        mouseSocket.disconnect();
+                        break;
+                    default:
+                        break;
+                }
+
+            }
+
+            @Override
+            public void onError(@NonNull String errorMessage) {
```

Comment:
We don't use @NonNull. Can you annotate with @NonNullByDefault on class level and use @Nullable where needed?